tests/test_transforms/test_transforms_processing.py

```diff
@@ -0,0 +1,177 @@
+        data_info = dict(
+            img=np.random.random((1333, 800, 3)),
+            gt_semantic_seg=np.random.random((1333, 800, 3)),
+            gt_bboxes=np.array([[0, 0, 112, 112]]),
+            gt_keypoints=np.array([[[20, 50, 1]]]))
+
+        with pytest.raises(AssertionError):
+            transform = Resize(scale=None, scale_factor=None)
+        with pytest.raises(TypeError):
+            transform = Resize(scale_factor=[])
+        # test scale is int
+        transform = Resize(scale=2000)
+        results = transform(copy.deepcopy(data_info))
+        assert results['img'].shape[:2] == (2000, 2000)
+        assert results['scale_factor'] == (2000 / 800, 2000 / 1333)
+
+        # test scale is tuple
+        transform = Resize(scale=(2000, 2000))
+        results = transform(copy.deepcopy(data_info))
+        assert results['img'].shape[:2] == (2000, 2000)
+        assert results['scale_factor'] == (2000 / 800, 2000 / 1333)
+
+        # test scale_factor is float
+        transform = Resize(scale_factor=2.0)
+        results = transform(copy.deepcopy(data_info))
+        assert results['img'].shape[:2] == (2666, 1600)
+        assert results['scale_factor'] == (2.0, 2.0)
+
+        # test scale_factor is tuple
+        transform = Resize(scale_factor=(1.5, 2))
+        results = transform(copy.deepcopy(data_info))
+        assert results['img'].shape[:2] == (2666, 1200)
+        assert results['scale_factor'] == (1.5, 2)
+
+        # test keep_ratio is True
+        transform = Resize(scale=(2000, 2000), keep_ratio=True)
+        results = transform(copy.deepcopy(data_info))
+        assert results['img'].shape[:2] == (2000, 1200)
+        assert results['scale'] == (1200, 2000)
+        assert results['scale_factor'] == (1200 / 800, 2000 / 1333)
+
+        # test resize_bboxes/seg/kps
+        transform = Resize(scale_factor=(1.5, 2))
+        results = transform(copy.deepcopy(data_info))
+        assert (results['gt_bboxes'] == np.array([[0, 0, 168, 224]])).all()
+        assert (results['gt_keypoints'] == np.array([[[30, 100, 1]]])).all()
+        assert results['gt_semantic_seg'].shape[:2] == (2666, 1200)
+
+        # test bbox_clip_border = False
+        data_info = dict(
+            img=np.random.random((300, 400, 3)),
+            gt_bboxes=np.array([[200, 150, 600, 450]]))
+        transform = Resize(scale=(200, 150), bbox_clip_border=False)
+        results = transform(data_info)
+        assert (results['gt_bboxes'] == np.array([100, 75, 300, 225])).all()
+
+    def test_repr(self):
+        transform = Resize(scale=(2000, 2000), keep_ratio=True)
+        assert repr(transform) == ('Resize(scale=(2000, 2000), '
+                                   'scale_factor=None, keep_ratio=True, '
+                                   'bbox_clip_border=True), backend=cv2), '
+                                   'interpolation=bilinear)')
+
+
+class TestPad:
+
+    def test_pad(self):
+        # test assertion
```

Comment:
assertion of what?
We can use `with self.assertRaisesRegex(AssertionError, 'xxx'):`

Reply:
This test is not a `TestCase` class, and doesn't have the `assertRaisesRegex` method.